### ISSUE TITLE
[Help Wanted] [PLT-7805] Hide integration message attachment text behind "Show More..." after it exceeds three lines of text on screen

### ISSUE THREAD

**Issue by esethna** [mattermost/mattermost-server#7582]
If you're interested please comment here and come [join our "Contributors" community channel](https://pre-release.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://pre-release.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://docs.mattermost.com/guides/developer.html), specifically for [machine setup](https://docs.mattermost.com/developer/developer-setup.html) and for [developer workflow](https://docs.mattermost.com/developer/developer-flow.html).

----

**Notes**: [Jira ticket](https://mattermost.atlassian.net/browse/PLT-7805)

Currently we cut the text after 300 characters, but this causes issues with Markdown rendering (e.g. code blocks and tables). See https://github.com/mattermost/mattermost-server/issues/7399.

Propose that instead of cutting off the text, we render the whole message attachment and use CSS to only show the first three lines on the screen until the user clicks "Show More...". This should apply for browsers, desktop apps and mobile apps.

Example:
1) Use this payload to post a webhook message
```
{"attachments":[{ "color": "#009edc", "text": "Build started by user.\n\n| Module | version | target env |\n| -------- | -------- | ------------------- |\n| Core | v2.3.0_27 | server1|\n| Batch | v2.3.0_16 | server1|\n" }]}
```

Observed:
![image](https://user-images.githubusercontent.com/13119842/31252461-48c66d10-a9ef-11e7-9a07-c5bac963ea0c.png)

Expected:
![image](https://user-images.githubusercontent.com/13119842/31252467-4ddcb610-a9ef-11e7-8ff9-9726a44e4de9.png)

**Comment by narendravardi** [comment]
@esethna can i work on this?

**Comment by esethna** [comment]
@narendravardi go for it, thanks!

**Comment by narendravardi** [comment]
@esethna, I was looking out for Java Repo and by mistake looked into this repo. I am not aware of Go and hence it would good if you could assign this to someone else.
Sorry! :)

**Comment by jasonblais** [comment]
@narendravardi Actually, this ticket won't involve any Go, and the code changes would be done in the https://github.com/mattermost/mattermost-webapp repository.

Let me know if you'd still be interested to give it a try?

**Comment by narendravardi** [comment]
Don't know ReactJS too :|

**Comment by jasonblais** [comment]
@narendravardi No worries! Thanks for your interest regardless! :) 

If you know Javascript, we have a few help wanted tickets on our desktop repository: https://github.com/mattermost/desktop/issues?q=is%3Aissue+is%3Aopen+label%3A%22Help+Wanted%22

Otherwise, there are open Hacktoberfest issues from other projects that involve Java, which can be found here: https://github.com/search?l=Java&q=label%3Ahacktoberfest+state%3Aopen+type%3Aissue&type=Issues

**Comment by KishoreFartiyal** [comment]
I would like to work on this issue

**Comment by jasonblais** [comment]
Great, thanks @KishoreFartiyal!

**Comment by KishoreFartiyal** [comment]
@jasonblais , can you help me reproduce this issue? 
I need steps to reproduce this issue and where to find the issue in UI?

**Comment by jasonblais** [comment]
@KishoreFartiyal Have you sent webhook messages before? If so, there's an example in the issue description.

If you haven't I can give you a few extra steps to help you guide through it

**Comment by KishoreFartiyal** [comment]
@jasonblais No, I have never sent webhook messages before. It'll be helpful if you can provide few steps from the beginning to create the issue.

**Comment by jasonblais** [comment]
@KishoreFartiyal Here's sample steps:

1. Go to Main Menu > Integrations > Incoming Webhook and click Add Incoming Webhook
2. Enter any name and description
3. Select Off Topic channel and save
4. Copy the URL on the Setup Successful page
5. Install Postman App (for Chrome, Mac, or Windows): https://www.getpostman.com/apps
6. Launch Postman and near the new tab at the top, click the drop-down to change GET to POST
7. Paste the URL you copied in step 5 into the "Enter request URL" field (e.g. something like https://ci-linux-postgres.mattermost.com/hooks/gt5cfekbffb7dmttxjhc97dyno)
8. Click the "Body" tab, select "raw" radio button, and paste in the text below
```
{"attachments":[{ "color": "#009edc", "text": "Build started by user.\n\n| Module | version | target env |\n| -------- | -------- | ------------------- |\n| Core | v2.3.0_27 | server1|\n| Batch | v2.3.0_16 | server1|\n" }]}
```
9. Click the "Send" button (if you scroll down you'll see "ok" if it worked)
10. Go back to Mattermost, click to go back to the team, and view "Off Topic" channel

^Let me know if that helps?

**Comment by KishoreFartiyal** [comment]
@jasonblais thanks for your help ! 
I am able to follow the steps conveniently .

Let's come to the issue.
Currently, it displays correctly as specified in the expected screenshot.

Expected Behavior from the description of the issue:
![image](https://user-images.githubusercontent.com/13373142/32539343-1f3c88d4-c48f-11e7-961c-e98262fe246f.png)

Current behavior:
![current markdown](https://user-images.githubusercontent.com/13373142/32539294-f18c3998-c48e-11e7-847f-c40dca5ce024.jpg)

Kindly help what is actually expected

**Comment by jasonblais** [comment]
@KishoreFartiyal Hmm I see. I think https://github.com/mattermost/mattermost-server/pull/7707 submitted Oct 24th may have fixed that particular issue.

I didn't find any corner cases so this issue can be considered fixed. Would there be another help wanted ticket you'd be interested taking a look?

**Comment by KishoreFartiyal** [comment]
@jasonblais no problem, I would pick another issue to work on

**Comment by matejzero** [comment]
I just hit this bug today with Mattermost 5.1.1. It it suppose to be fixed?

**Comment by jasonblais** [comment]
Thanks @matejzero. Looks like I closed the ticket in error.

The issue is fixed in Mattermost 5.2, as part of https://mattermost.atlassian.net/browse/MM-11422

**Comment by matejzero** [comment]
That’s really good news. I guess this fix will arrive in september’s gitlab
release.

Thanks!

On 13 Sep 2018, at 15:55, Jason Blais <notifications@github.com> wrote:

Thanks @matejzero <https://github.com/matejzero>. Looks like I closed the
ticket in error.

The issue is fixed in Mattermost 5.2, as part of
https://mattermost.atlassian.net/browse/MM-11422

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/mattermost/mattermost-server/issues/7582#issuecomment-421016013>,
or mute the thread
